Public/Src/Cache/ContentStore/Interfaces/Results/PutResult.cs

```diff
@@ -15,12 +15,13 @@ public class PutResult : BoolResult, IEquatable<PutResult>
         /// <summary>
         ///     Initializes a new instance of the <see cref="PutResult" /> class.
         /// </summary>
-        public PutResult(ContentHash contentHash, long contentSize)
+        public PutResult(ContentHash contentHash, long contentSize, bool contentAlreadyExistsInCache = false)

```

Comment:
This is the break-risking change that Sergey warned about. If you add this optional argument, then customers (e.g. ADO) that link to us and use the constrcutor will break unless they re-compile against the new bits. To avoid this, you need to add a separate new constructor.